client/src/util/testScheduler.ts

```diff
@@ -121,3 +124,58 @@ export const filterHelperMissingCourses = (combination: CourseSection[]): boolea
     }
     return true;
 };
+
+/**
+ * Removes schedules that have overlapping times
+ * @param {CourseSection[][]} combinations generated course schedules
+ * @returns {CourseSection[][]} returns filtered version of combinations
+ */
+export const filterCombinationsWithTimeOverlaps = (combinations: CourseSection[][]): CourseSection[][] => {
+    return combinations.filter((combination: CourseSection[]) => filterHelperTimeOverlaps(combination));
+};
+
+/**
+ * Helper function for filterCombinationsWithTimeOverlaps
+ * @param {CourseSection[]} combination a course schedule
+ * @returns {boolean} returns false if the given schedule should be filtered out (ie: there are time overlaps)
+ */
+export const filterHelperTimeOverlaps = (combination: CourseSection[]): boolean => {
+    // NOTE: disabling lint rule here because i don't know how to do this yet without the indices in the for loops
+    // eslint-disable-next-line @typescript-eslint/prefer-for-of
+    for (let i = 0; i < combination.length; i++) {
+        for (let j = i + 1; j < combination.length; j++) {
```

Comment:
I just realized I should probably be checking if the course even has starttime and endtime, I think some courses have nothing. :(

Reply:
would it make sense to treat courses that don't have times separately from those that do? most of the no-time courses are loosely structured like directed studies or seminars so they would never show on worklists anyway, i think

Reply:
True, they wouldnt need to be filtered out since they won't overlap with anything. I was thinking we can skip if any course doesnt have a time.

Reply:
oh duh... yeah that sounds a lot simpler LOL

Reply:
Okay done!